### PR TITLE
DOC: tutorial updates

### LINKED ISSUE #36: tutorial modifications
- [x] update name of doc file from `methods.md` to `tutorial.md`
- [x] break commands over multiple lines, like in the Q2 docs
- [x] add tutorial data to `Snakemake` folder, update instructions for accessing that data
- ~[ ] _Note however that usually you would perform some manual filtering and trimming between these two steps, so these two commands likely won’t get you a publication quality phylogeny._ : note that we are working on adding more support of these steps~ this is now part of #67
- [x] Add suggestions on modifications that need to be made to Snakemake file to apply to non-tutorial data (e.g., update file names and `N_THREADS`)

### LINKED ISSUE #64: add documentation section on running in parallel
I've received questions from multiple users on how to run in parallel, so we should add a specific section to the docs on this. Here is some text copied from my replies that we can use in this section:

genome-sampler can be run in parallel to speed it up. This is done in different ways depending on whether you're running the steps individually or through Snakemake. 

If you're using Snakemake, you need to edit `Snakefile` and set the `N_THREADS` value to the number of threads you'd like genome-sampler to use. 

If you're running the steps individually you can pass the `--p-n-threads` option to several of the commands. For example, `sample-diversity` is the slowest step in the workflow. You can provide the `--p-n-threads` parameter to run it in parallel:

```
qiime genome-sampler sample-diversity \
 --i-context-seqs filtered-context-seqs.qza \
 --p-percent-id 0.9995 \
 --o-selection diversity-selection.qza \
 --p-n-threads n
```

When running this command, you should set n to be the number of available processors or cores on a single node of your system. For example, I work on a cluster that has nodes with 28 cores, so when I submit this job I would run:

```
qiime genome-sampler sample-diversity \
 --i-context-seqs filtered-context-seqs.qza \
 --p-percent-id 0.9995 \
 --o-selection diversity-selection.qza \
 --p-n-threads 28
```

This uses all of the resources on a single node of the cluster for me. In the future we'll be adding support for splitting workflows like this across multiple cluster nodes, but we do not have this support at this time.

### LINKED ISSUE #66: illustrate sampling of focal sequences in tutorial
This came up in [peer review](https://f1000research.com/articles/9-657/v1#referee-response-65756).

### LINKED ISSUE #67: illustrate alignment, masking, and tree building in tutorial
Include any discussion of limitations.

### PR DESCRIPTION
**Pull request by gregcaporaso**
This PR expands the genome-sampler documentation and splits the docs into a multi-chapter JupyterBook. The expansion addresses the following issues:
 - fixes #67
 - fixes #66
 - fixes #64
 - fixes #36

Before merge:
 - ~document tree rooting~ (we're going to skip this for this release)

### CODE REVIEW

**Comment by thermokarst** [comment]
Another potential TODO: update installation guide to use packages.qiime2.org

**Comment by gregcaporaso** [comment]
@thermokarst, I created a separate issue for your install note (#73).

File: docs/other.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+`filtered-context-seqs.qza` file.
+
+
+Alternatively, if you want to remove these sequences from your focal sequence
+collection, you could do this as follows:
+
+```
+qiime feature-table filter-seqs\
+ --i-data focal-seqs.qza \
+ --m-metadata-file context-seqs.qza \
```
Comment (ebolyen) [inline docs/other.md:141]: So I have learned from the benchmarking that this in practice make provenance absolutely massive. We might want to find a new approach here.

File: docs/other.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+metadata. All of your downstream analysis steps should then use the resulting
+`filtered-context-seqs.qza` file.
+
+Alternatively, if you want to remove sequences from your focal sequence
+collection instead of from your context sequence collection, you can edit your
+focal sequence metadata file, and then use that to filter in the same way that
+context sequences were filtered here.
+
+```{note}
+If your metadata file already contains a column that differeniates the
```
Comment (ebolyen) [inline docs/other.md:178]:
```suggestion
If your metadata file already contains a column that differentiates the
```

File: docs/other.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+(parallel)=
+## Running in parallel
+
+`genome-sampler` can be run in parallel to speed it up. This is done in
+different ways depending on whether you're running the steps individually or
+through Snakemake.
+
+If you're using Snakemake, you need to edit `Snakefile` to set the `N_THREADS`
+value to the number of threads you'd like genome-sampler to use.
```
Comment (ebolyen) [inline docs/other.md:23]: placeholder note that this will change.

File: docs/tutorial.md [hunk 4]
```diff
@@ -1,60 +1,33 @@
...
-to get started quickly and use default parameters, start here. If you'd like
-more control over your analysis or want to work through the steps
+The full `genome-sampler` workflow can be run using
+[Snakemake](https://snakemake.readthedocs.io/en/stable/)
+{cite}`snakemake-bioi`. If you'd like to get started quickly and use default
+parameters, start here.
+If you'd like more control over your analysis or want to work through the steps
 individually, move on to the next section.
 
 Download the Snakemake file using `curl` as follows:
```
Comment (ebolyen) [inline docs/tutorial.md:33]: placeholder note indicating the config will be needed also.

File: docs/other.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+This would use all of the resources on a single node of that cluster. In the
+future QIIME 2 will have support for splitting workflows like this across
+multiple cluster nodes, but we do not have this support now.
+
+(adapting-tutorial)=
+## Adapting Snakemake workflow for application to your own data
+The Snakemake workflow presented in the tutorial is a good starting point for
+your own analyses. There are typically a few things to do to adapt
+`Snakefile` for your own data. The changes that you'll make to the `Snakefile`
+are all in the section at the top that is annotated as ``CONFIGS``.
```
Comment (ebolyen) [inline docs/other.md:59]: note that this will change

File: docs/other.md (hunk 1 above)
Comment (gregcaporaso) [reply docs/other.md]: Fixed this by updated to pass metadata as metadata instead of sequences as metadata.

**Comment by ebolyen** [comment]
I am going to merge this so that I can follow the docs build on master and make sure everything works. (Too many URLs 404 to be worth testing outside of master.)

**Pull request merged**